`Debug` not implemented for `HkdfExtract<Sha256>` or `Hkdf<Sha256>`

Here’s a minimal test case:

```rust
#[test]
fn test_debug() {
    fn check_debug<T: core::fmt::Debug>() {}

    check_debug::<HkdfExtract<Sha256>>();
    check_debug::<Hkdf<Sha256>>();
}
```

It fails with this error:

```text
error[E0277]: the trait bound `CoreWrapper<CtVariableCoreWrapper<Sha256VarCore, UInt<UInt<UInt<UInt<UInt<UInt<UTerm, B1>, B0>, B0>, B0>, B0>, B0>, OidSha256>>: AlgorithmName` is not satisfied
   --> hkdf/tests/tests.rs:388:19
    |
388 |     check_debug::<HkdfExtract<Sha256>>();
    |                   ^^^^^^^^^^^^^^^^^^^ the trait `AlgorithmName` is not implemented for `CoreWrapper<CtVariableCoreWrapper<Sha256VarCore, UInt<UInt<UInt<UInt<UInt<UInt<UTerm, B1>, B0>, B0>, B0>, B0>, B0>, OidSha256>>`
    |
    = help: the following other types implement trait `AlgorithmName`:
              HmacCore<D>
              Sha1Core
              Sha256VarCore
              Sha512VarCore
              CtVariableCoreWrapper<T, OutSize, O>
    = note: required for `HkdfExtract<CoreWrapper<CtVariableCoreWrapper<Sha256VarCore, UInt<UInt<UInt<UInt<UInt<UInt<UTerm, B1>, B0>, B0>, B0>, B0>, B0>, OidSha256>>>` to implement `Debug`
note: required by a bound in `check_debug`
   --> hkdf/tests/tests.rs:386:23
    |
386 |     fn check_debug<T: core::fmt::Debug>() {}
    |                       ^^^^^^^^^^^^^^^^ required by this bound in `check_debug`

error[E0277]: the trait bound `CoreWrapper<CtVariableCoreWrapper<Sha256VarCore, UInt<UInt<UInt<UInt<UInt<UInt<UTerm, B1>, B0>, B0>, B0>, B0>, B0>, OidSha256>>: AlgorithmName` is not satisfied
   --> hkdf/tests/tests.rs:389:19
    |
389 |     check_debug::<Hkdf<Sha256>>();
    |                   ^^^^^^^^^^^^ the trait `AlgorithmName` is not implemented for `CoreWrapper<CtVariableCoreWrapper<Sha256VarCore, UInt<UInt<UInt<UInt<UInt<UInt<UTerm, B1>, B0>, B0>, B0>, B0>, B0>, OidSha256>>`
    |
    = help: the following other types implement trait `AlgorithmName`:
              HmacCore<D>
              Sha1Core
              Sha256VarCore
              Sha512VarCore
              CtVariableCoreWrapper<T, OutSize, O>
    = note: required for `Hkdf<CoreWrapper<CtVariableCoreWrapper<Sha256VarCore, UInt<UInt<UInt<UInt<UInt<UInt<UTerm, B1>, B0>, B0>, B0>, B0>, B0>, OidSha256>>>` to implement `Debug`
note: required by a bound in `check_debug`
   --> hkdf/tests/tests.rs:386:23
    |
386 |     fn check_debug<T: core::fmt::Debug>() {}
    |                       ^^^^^^^^^^^^^^^^ required by this bound in `check_debug`

For more information about this error, try `rustc --explain E0277`.
```

Any ideas?

(Edited with a more concise test case and error message.)

--- a/hkdf/Cargo.toml
+++ b/hkdf/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "hkdf"
-version = "0.12.3" # Also update html_root_url in lib.rs when bumping this
+version = "0.12.4"
 authors = ["RustCrypto Developers"]
 license = "MIT OR Apache-2.0"
 homepage = "https://github.com/RustCrypto/KDFs/"

--- a/hkdf/src/lib.rs
+++ b/hkdf/src/lib.rs
@@ -89,8 +89,7 @@
 #![no_std]
 #![doc(
     html_logo_url = "https://raw.githubusercontent.com/RustCrypto/media/6ee8e381/logo.svg",
-    html_favicon_url = "https://raw.githubusercontent.com/RustCrypto/media/6ee8e381/logo.svg",
-    html_root_url = "https://docs.rs/hkdf/0.12.3"
+    html_favicon_url = "https://raw.githubusercontent.com/RustCrypto/media/6ee8e381/logo.svg"
 )]
 #![cfg_attr(docsrs, feature(doc_cfg))]
 #![forbid(unsafe_code)]
@@ -173,12 +172,13 @@ where
 
 impl<H, I> fmt::Debug for HkdfExtract<H, I>
 where
-    H: OutputSizeUser + AlgorithmName,
+    H: OutputSizeUser,
     I: HmacImpl<H>,
+    I::Core: AlgorithmName,
 {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         f.write_str("HkdfExtract<")?;
-        <H as AlgorithmName>::write_alg_name(f)?;
+        <I::Core as AlgorithmName>::write_alg_name(f)?;
         f.write_str("> { ... }")
     }
 }
@@ -273,12 +273,13 @@ impl<H: OutputSizeUser, I: HmacImpl<H>> Hkdf<H, I> {
 
 impl<H, I> fmt::Debug for Hkdf<H, I>
 where
-    H: OutputSizeUser + AlgorithmName,
+    H: OutputSizeUser,
     I: HmacImpl<H>,
+    I::Core: AlgorithmName,
 {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         f.write_str("Hkdf<")?;
-        <H as AlgorithmName>::write_alg_name(f)?;
+        <I::Core as AlgorithmName>::write_alg_name(f)?;
         f.write_str("> { ... }")
     }
 }

--- a/hkdf/tests/tests.rs
+++ b/hkdf/tests/tests.rs
@@ -446,3 +446,10 @@ new_test!(
     "wycheproof-sha512",
     SimpleHkdf::<Sha512>
 );
+
+#[test]
+fn test_debug_impls() {
+    fn needs_debug<T: std::fmt::Debug>() {}
+    needs_debug::<Hkdf<Sha256>>();
+    needs_debug::<HkdfExtract<Sha256>>();
+}

